src/command_modules/azure-cli-extension/azure/cli/command_modules/extension/__init__.py

```diff
@@ -60,6 +60,16 @@ def load_arguments(self, command):
             c.argument('extension_name', completer=extension_name_from_index_completion_list)
             c.argument('source', options_list=['--source', '-s'], help='Filepath or URL to an extension', completer=FilesCompleter())
             c.argument('yes', options_list=['--yes', '-y'], action='store_true', help='Do not prompt for confirmation.')
+            c.argument('extra_index_urls', options_list=['--extra-index-url'], action='append',
+                       help='Extra URLs of package indexes to use. This should point to a repository compliant '
+                            'with PEP 503 (the simple repository API) or a local directory laid out in the same '
+                            'format. Similar to the equivalent flag in pip install.')
+
+        with self.argument_context('extension update') as c:
+            c.argument('extra_index_urls', options_list=['--extra-index-url'], action='append',
```

Comment:
nargs='+' instead of action='append'

They are similar but slightly different.

https://docs.python.org/3/library/argparse.html#nargs
https://docs.python.org/3/library/argparse.html#action

Reply:
Please correct me if I am wrong. But, I think action='append' is what we need. I am assuming the command to be like "az extension add -s <extension_url> --extra-index-url <URL1> --extra-index-url <URL2> --extra-index-url <URL3>"

So, for the above command I think action='append' is suitable. If we add nargs='+' then that would mean
"az extension add -s <extension_url> --extra-index-url <URL1> <URL2> <URL3>"

I chose the former one because that is the pip install semantics.  But, I have limited experience in python argparser so I might be wrong.

Reply:
Got your point. By nargs='+', we will be able to support "az extension add -s <extension_url> --extra-index-url <URL1> <URL2> <URL3>", which is better than "az extension add -s <extension_url> --extra-index-url <URL1> --extra-index-url <URL2> --extra-index-url <URL3>"